### ISSUE TITLE
NativeScript SQLite dialect

### ISSUE THREAD

**Issue by darkyelox** [sequelize/sequelize#11261]
**Is your feature request related to a problem? Please describe.**
Well, not a problem with this library but one from TypeORM that have a basic or some lack of support in NativeScript framework, and i see a low work in the library it self so i'm searching for ORM alternatives to my apps and this one looks promising.

**Describe the solution you'd like**
I know you have already a dialect for SQLite but i propose to create one for NativeScript like using the already existent plugin "nativescript-sqlite" as wrapper, maybe use the same code from the SQLite dialect folder but calling this library on in instead of the native SQLite library, and define it in the connection:

```javascript
const sequelize = new Sequelize('database', 'username', 'password', {
  // sqlite! now!
  dialect: 'nativescript-sqlite',

  // the storage engine for sqlite
  // - default ':memory:'
  storage: 'database.sql'
})
```

**Why should this be in Sequelize**
I think it will not be hard to implement and allow this library jump to mobile development as a "must to use ORM" for bridge applications.

**Describe alternatives you've considered**
No one but keep using TypeORM

**Comment by papb** [comment]
Hello! Thanks for opening the issue. I am not familiar with NativeScript, but after a brief search I found [an official guide on how to use (almost?) any npm module with it](https://www.nativescript.org/blog/how-to-use-any-npm-module-with-nativescript).

Have you tried that? If yes, can you please post what exactly was the problem? That page mentions that some NodeJS-specific libraries could cause problems, such as `fs` and such... Is running SQLite itself the problem?

In other words: are you able to use Sequelize with another dialect and this is just a sqlite-specific issue, or you are unable to use Sequelize at all?

Hopefully I can work with you to find a way to make it work... (I might say some dumb things though because I know nothing about NativeScript... Sorry in advance...)

**Comment by darkyelox** [comment]
@papb don't worry, i'll explain to you how NativeScript works:

It's a mobile framework that use engines for each platform (Android and iOS) for run javascript code over it but using native components for the UI (so you write a `<Button>` tag in a html file and it's mapped to a native Android Button and a UIButton for iOS), is like a NodeJS platform but have some limitations like you mentioned with `fs` packages, maybe i can use sequelize for connecting to a remote MySQL database without problems but SQLite is always a local connection and we can't use the native sqlite NPM package because of the implementation of NativeScript, for that reason there is a plugin, [nativescript-sqlite](https://github.com/NathanaelA/nativescript-sqlite) is intended for access SQLite database files from mobile devices independent of the platform (Android, iOS).

So the best way for use this library with NativeScript (like TypeORM does it) is create a new dialect, maybe a copy/paste of the already existent sqlite dialect but using the nativescript-sqlite plugin behind (at the end the plugin is also a NPM package) instead of sqlite NPM package.

We have also have to pay attention on the imports of the library for don't use any browser based import on it because that also breaks NativeScript (like use `document` object or specific objects for web browsers).

Then, you can write a pure javascript / typescript dialect as normal but import the nativescript-sqlite package functions and use it for do queries, statements, migrations, etc.

**Comment by papb** [comment]
Hmm... I see... Thanks for the explanation.

The [Sequelize constructor](https://sequelize.org/master/class/lib/sequelize.js~Sequelize.html#instance-constructor-constructor) has options to change the dialect module used under the hood: `options.dialectModule` and `dialectModulePath` - have you tried those?

**Comment by darkyelox** [comment]
Oh, i didn't see that options, maybe I'll try them this weekend but i don't have too much hope, let's see what happens, thanks for give me that points, meanwhile this issue will stay open.
